subseas_flex_fcst/predictions.py

```diff
@@ -69,3 +69,19 @@ def combine_cptdataset(dataDir,filePattern,dof=False):
         dataDic[idx] = ds
     dataCombine = xr.combine_by_coords([dataDic[x] for x in dataDic],combine_attrs="drop_conflicts")
     return dataCombine
+
+
+def cpt_startsList(dataPath,filePattern,dateSearchPattern,zeroPadding=False):
+    filesNameList = glob.glob(f'{dataPath}/{filePattern}')
+    startDates = []
+    for idx, i in enumerate(filesNameList):
+        startDate = re.search(dateSearchPattern,filesNameList[idx])
+        startDatedt = datetime.strptime(startDate.group(),"%b-%d-%Y")
+        startDateStr = startDatedt.strftime("%b-%-d-%Y")
+        startDates.append(startDatedt)
+    startDates = sorted(set(startDates))
+    if zeroPadding == False:
+        startDates = [i.strftime("%b-%-d-%Y") for i in startDates] #needs to have date with no zero padding to match the file path namesme
+    if zeroPadding == True:
+        startDates = [i.strftime("%b-%d-%Y") for i in startDates]
```

Comment:
I don't think zeroPadding is going to help many cases. Some files will have 0s, some don't, that's covered. Some files will have the month expressed in numbers, with or without leading 0s... and it can go on like this. I would put the input of strftime as an input of the function instead.

Reply:
In light of previous comments above. There might be a need to have a set of functions that manipulate time. We need:

1. functions that are given the strftime format of file names and convert that in datetime objects;
2. functions that manipulate the different datetime objects, for instance to retrieve target_start and target_end from issue and lead
3. functions that print our datetime objects in strftime formats we want in the app at various locations.

I am not pretending I have the solution but hopefully it helps rationalize the whole set.

Reply:
Yes I think adding the zeroPadding option was my first attempt at trying to allow some flexibility in the way the user can handle the way the dates are output. Then I hoped by having the user be able to input the search pattern for the date that would handle at least flexibility in the way the dates are expressed initially in the file names when the search occurs (though this does assume the user knows enough about regex to figure out the pattern- though we can give a few common examples in the docs if this function gets approved.)

Maybe it is worth making an issue to address a set of functions that manipulate time. And speak a bit more about what functions might make the most sense for this.

I guess this all might not answer the question of if I should remove the zero_padding option from the file though.

Reply:
ok with setting up an issue

Reply:
I made a new issue to address making new functions, and changed the zero_padding option for the existing function to take the strftime format as an input.